src/main/java/org/opensearch/sdk/sample/helloworld/rest/RestHelloAction.java

```diff
@@ -50,13 +52,18 @@ public class RestHelloAction extends BaseExtensionRestHandler {
     private List<String> worldAdjectives = new ArrayList<>();
     private Random rand = new Random();
 
+    /**
+     * Instantiate this action
+     */
+    public RestHelloAction() {}
+
     @Override
     public List<RouteHandler> routeHandlers() {
         return List.of(
-            new RouteHandler(GET, "/hello", handleGetRequest),
-            new RouteHandler(POST, "/hello", handlePostRequest),
-            new RouteHandler(PUT, "/hello/{name}", handlePutRequest),
-            new RouteHandler(DELETE, "/goodbye", handleDeleteRequest)
+            new ExtensionRouteHandler("greet", GET, "/hello", handleGetRequest),
```

Comment:
This appears to be the only case where we are actually using the new `ExtensionRouteHandler` class.

This method signature is an override that only exists if we are extending `BaseExtensionRestHandler` so it seems to me we could add all this logic in that superclass.   The `BaseExtensionRestHandler` class could fetch the shortname from the settings and then just expose a `protected routePrefix(String name)` method that returns "shortname:name".  Then this line would just be `new RouteHandler(routePrefix("greet"), GET, ...)` and you'd save a ton of intermediate objects and initialization checking, etc.